File: app/src/test/java/com/mxt/anitrend/util/EpisodeUtil_GetActualTitleTest.java
package com.mxt.anitrend.util;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.equalTo;

@RunWith(Parameterized.class)
public class EpisodeUtil_GetActualTitleTest {

    @Parameterized.Parameters
    public static Iterable<Object[]> data() {
        return Arrays.asList(new Object[][]{
                {"Boku no Hero Academia - Episode 23", "Boku no Hero Academia"},
                {"Haikyuu Season 3", "Haikyuu"},
                {"Boku no Hero Academia Season 2 - Episode 19", "Boku no Hero Academia"}
        });
    }

    @Parameterized.Parameter(0)
    public String inputTitle;

    @Parameterized.Parameter(1)
    public String actualTitle;


    @Test
    public void getActualTile() {
        assertThat(EpisodeUtil.getActualTile(inputTitle), equalTo(actualTitle));
    }

}

File: app/src/test/java/com/mxt/anitrend/util/EpisodeUtil_GetActualTitleTest.kt
package com.mxt.anitrend.util

import com.mxt.anitrend.util.collection.EpisodeUtil
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.equalTo
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class EpisodeUtil_GetActualTitleTest(
    private val inputTitle: String,
    private val actualTitle: String
) {

    companion object {
        @JvmStatic
        @Parameterized.Parameters
        fun data(): Iterable<Array<String>> = listOf(
            arrayOf("Boku no Hero Academia - Episode 23", "Boku no Hero Academia"),
            arrayOf("Haikyuu Season 3", "Haikyuu"),
            arrayOf("Boku no Hero Academia Season 2 - Episode 19", "Boku no Hero Academia")
        )
    }

    @Test
    fun getActualTile() {
        assertThat(EpisodeUtil.getActualTile(inputTitle), equalTo(actualTitle))
    }
}
